controllers/controllers/resource/template.go

```diff
@@ -76,23 +76,40 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		controlPlaneTemplateName = cp.Spec.MachineTemplate.InfrastructureRef.Name
 	}
 
+	kubeadmconfigTemplateNames := make(map[string]string, len(clusterSpec.Spec.WorkerNodeGroupConfigurations))
 	workloadTemplateNames := make(map[string]string, len(clusterSpec.Spec.WorkerNodeGroupConfigurations))
 	for _, workerNodeGroupConfiguration := range clusterSpec.Spec.WorkerNodeGroupConfigurations {
-		oldVmc, err := r.ExistingVSphereWorkerMachineConfig(ctx, eksaCluster, workerNodeGroupConfiguration)
+		existingKubeadmConfigTemplate, err := r.KubeadmConfigTemplate(ctx, eksaCluster, workerNodeGroupConfiguration)
```

Comment:
do u think if we can keep `KubeadmConfigTemplate` and `MapKubeadmConfigTemplateToWorkerNodeGroupConfiguration` in the fetcher.go, and just call a method like `ExistingWorkerNodeGroupConfig` to get the `oldWn`? just to keep it consistent w the rest of the code in this method

Reply:
yeah we should be able to do that

Reply:
https://github.com/aws/eks-anywhere/pull/1175/commits/0697232a99fef364b34fcdd25c784bac4473c2f2